src/sentry/api/serializers/rest_framework/group_notes.py

```diff
@@ -1,7 +1,22 @@
 from __future__ import absolute_import
 
 from rest_framework import serializers
+from sentry.models import Project
+
+from .list import ListField
 
 
 class NoteSerializer(serializers.Serializer):
     text = serializers.CharField()
+    mentions = ListField(required=False)
+
+    def validate_mentions(self, attrs, source):
+        if source in attrs and 'group' in self.context:
+            mentions = attrs[source]
+            project = Project.objects.get(id=self.context['group'].project_id)
```

Comment:
this can just be `project = group.project` (and I think based on `GroupEndpoint`, that'll actually save a query)